packages/pluggableWidgets/fileuploader-web/src/stores/FileUploaderStore.ts

```
@@ -0,0 +1,153 @@
+                    this.files.push(FileStore.existingFile(item, this));
+
+                    this.lastSeenItems.add(item.id);
+                }
+
+                this.existingItemsLoaded = true;
+            }
+        } else {
+            // todo: re-check that existing items are still there
+            const newItems = findNewItems(this.lastSeenItems, itemsDs.items || []);
+            if (newItems) {
+                for (const newItem of newItems) {
+                    const obj = extractMxObject(newItem);
+                    if (!obj.get2("HasContents")) {
+                        // todo resolve waiting
+                        const firstWaiting = this.currentWaiting.shift();
+                        if (firstWaiting) {
+                            firstWaiting(newItem);
+                        }
+                    } else {
+                        // adding this file as file possibly created by someone else
+                        this.files.push(FileStore.existingFile(newItem, this));
+                    }
+
+                    this.lastSeenItems.add(newItem.id);
+                }
+            }
+        }
+    }
+
+    get canRequestFile(): boolean {
+        return this.existingItemsLoaded && !!this._createFileAction;
+    }
+
+    requestFileObject(): Promise<ObjectItem> {
+        if (!this.canRequestFile) {
+            throw new Error("Can't request file");
+        }
+
+        return new Promise<ObjectItem>(resolve => {
+            this._createFileAction!.execute();
+
+            this.currentWaiting.push(resolve);
+        });
+    }
+
+    async processDrop(acceptedFiles: File[], fileRejections: FileRejection[]): Promise<void> {
+        if (!this._createFileAction || !this._createFileAction.canExecute) {
+            console.warn(
+                `'Action to create new files' is not available or can't be executed. Please check if '${this._widgetName}' widget is configured correctly.`
+            );
```

Should it be en error as well? I think Aciton is required and we should "fail fast" if it's not present.
Alos, I think this should be part of consistency check.

---

I added this as part of consistency errors, good point! Let's keep this one for the case where it is not available for some reason, as this situation can happen dynamically.